src/components/lib/ConnectWalletDialog.js

```diff
@@ -3,124 +3,197 @@ import Check from '../svgs/Check'
+	pollingInterval: 1000,
+	requestTimeoutMs: 1000,
+	config: {
+		chainId: CHAIN_ID,
+	},
+	url: ETH_RPC_URL,
+	manifestEmail: 'contact@keep.network', 
+	manifestAppUrl: 'https://localhost'
 })
 
 // Wallets.
 const WALLETS = [
 	{
 		name: "Metamask",
 		icon: "/images/metamask-fox.svg",
-		showName: true,
 		connector: injectedConnector
 	},
 	{
-		name: "Ledger",
+		name: "Ledger Legacy",
 		icon: "/images/ledger.svg",
-		connector: ledgerConnector
+		connector: ledgerLegacyConnector,
+		isHardwareWallet: true,
 	},
+	{
+		name: "Ledger Live",
+		icon: "/images/ledger.svg",
+		connector: ledgerLiveConnector,
+		isHardwareWallet: true,
+	},
+	{
+		name: "Trezor",
+		icon: "/images/trezor.svg",
+		connector: trezorConnector,
+		isHardwareWallet: true,
+	}
 ]
 
+
 export const ConnectWalletDialog = ({ shown, onConnected, onClose }) => {
-	const { active, account, activate, chainId, connector } = useWeb3React()
+	const { active, account, activate } = useWeb3React()
 
-	let [chosenWallet, setChosenWallet] = useState(null)
+	let [chosenWallet, setChosenWallet] = useState({})
 	let [error, setError] = useState(null)
-	let state = {
-		chosenWallet,
-		error
-	}
+	const [availableAccounts, setAvailableAccounts] = useState([])
 
 	async function chooseWallet(wallet) {
-		setChosenWallet(wallet)
+		try {
+			setChosenWallet(wallet)
+			if(wallet.isHardwareWallet) {
+				await wallet.connector.activate()
+				setAvailableAccounts(await wallet.connector.getAccounts())
+			} else {
+				await activateProvider(null, wallet)
+			}
+		} catch(error) {
+			setError(error.toString())
+		}
+	}
 
+	const activateProvider = async (selectedAccount, wallet = chosenWallet) => {
 		try {
+			if(wallet.isHardwareWallet) {
+				wallet.connector.setDefaultAccount(selectedAccount)
+			}
 			await activate(wallet.connector, undefined, true)
 			onConnected()
 		} catch(ex) {
 			setError(ex.toString())
-			throw ex
 		}
 	}
 
-	const ChooseWalletStep = () => {
-		return <>
-			<div className="title">Connect to a wallet</div>
-			<p>This wallet will be used to sign transactions on Ethereum.</p>
- 
-			<ul className='wallets'>
-				{
-					WALLETS.map(wallet => {
-						return <li className='wallet-option' onClick={() => chooseWallet(wallet)}>
-							<img src={wallet.icon} />
-							{wallet.showName && wallet.name}
-						</li>
-					})
-				}
-			</ul>
-		</>
+	const reconnectWallet = async () => {
+		setError(null)
+		await chooseWallet(chosenWallet)
 	}
 
-	const ConnectToWalletStep = () => {
-		if(error) {
-			return <ErrorConnecting/>
-		}
+	return <div className={`modal connect-wallet ${shown ? 'open' : 'closed'}`}>
+		<div className="modal-body">
+			<div className="close">
+				<div className="x" onClick={onClose}>&#9587;</div>
+			</div>
+			{!chosenWallet.name && <ChooseWalletStep onChooseWallet={chooseWallet} />}
+			{(chosenWallet.name && !active) &&
+				<ConnectToWalletStep
+				wallet={chosenWallet}
+				error={error}onTryAgainClick={reconnectWallet}
```

Comment:
Spacing and indentation is odd here but we can fix it when we do a pass with prettier